### PR TITLE
fix: disabled the button while it's loading, so it doesn't add the same repo again

### LINKED ISSUE #1507: Bug: create insight page allowing to add same repo
### Describe the bug

create insight page allowing to add same repo, it would be great if we could check it. 

https://github.com/open-sauced/insights/assets/18273833/5ffd0f59-718b-4f4b-a7b2-16dd17b62bb3

### Steps to reproduce

1. go to https://insights.opensauced.pizza/hub/insights/new
2. add a repo (if you click twice it will be added twice)
3. remove it from the added repos 
4. add it again from the suggested repos 
5. you can find it added twice

### Browsers

_No response_

### Additional context (Is this in dev or production?)

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [ ] I agree to follow this project's Contribution Docs

### PR DESCRIPTION
**Pull request by deepakrudrapaul**
## Description

This PR disabled the button while it's loading, so it doesn't add the same repo again

## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [x] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents

Fixes #1507 

## Mobile & Desktop Screenshots/Recordings

Before:
before-repo

After:
after-repo

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [x] 🙅 no documentation needed

## [optional] What gif best describes this PR or how it makes you feel?
:smiley:

### CODE REVIEW

File: components/molecules/SuggestedRepo/suggested-repo.tsx [hunk 1]
```diff
@@ -4,12 +4,22 @@ import RepoCardProfile, { RepoCardProfileProps } from "../RepoCardProfile/repo-c
...
+const SuggestedRepository = ({ data, onAddRepo, loadingData }: SuggestedRopsitoryProps) => {
   return (
     <div className="flex items-center justify-between">
       <RepoCardProfile {...data} />
       <Button
+        disabled={
+          `${data.orgName}/${data.repoName}` === loadingData?.repoName &&
+          loadingData?.isLoading &&
+          !loadingData?.isAddedFromCart
+        }
```
Comment (babblebey) [inline components/molecules/SuggestedRepo/suggested-repo.tsx:22]:
```suggestion
        disabled={
        repos.length === insightRepoLimit ||
          (addRepoLoading.repoName === repoSearchTerm && addRepoLoading.isLoading)
        }
        loading={
        repos.length === insightRepoLimit ||
          (addRepoLoading.repoName === repoSearchTerm && addRepoLoading.isLoading)
        }
```

File: components/organisms/InsightPage/InsightPage.tsx [hunk 2]
```diff
@@ -396,13 +399,21 @@ const InsightPage = ({ edit, insight, pageRepos }: InsightPageProps) => {
           />
 
           <div>
-            <Button disabled={repos.length === insightRepoLimit} onClick={handleAddRepository} variant="outline">
+            <Button
+              disabled={
+                repos.length === insightRepoLimit ||
+                (addRepoLoading.repoName === repoSearchTerm && addRepoLoading.isLoading)
+              }
```
Comment (babblebey) [inline components/organisms/InsightPage/InsightPage.tsx:406]:
```suggestion
              disabled={
              repos.length === insightRepoLimit ||
                (addRepoLoading.repoName === repoSearchTerm && addRepoLoading.isLoading)
              }
              loading={
              repos.length === insightRepoLimit ||
                (addRepoLoading.repoName === repoSearchTerm && addRepoLoading.isLoading)
              }
```

**Review by babblebey — COMMENTED** [review]
Great stuff @deepakrudrapaul 

In addition to this, I'd suggest you leverage the `loading` prop on the `Button` component to further inform the User of the a loading coupled with the disabling.

It'll only complete the user experience.

File: components/molecules/SuggestedRepo/suggested-repo.tsx (hunk 1 above)
Comment (deepakrudrapaul) [reply components/molecules/SuggestedRepo/suggested-repo.tsx:22]: Thank you for the thoughtful suggestion!

Comment (deepakrudrapaul) [reply components/molecules/SuggestedRepo/suggested-repo.tsx:22]: Thank you for the thoughtful suggestion!

**Comment by deepakrudrapaul** [comment]
> Great stuff @deepakrudrapaul 
> 
> In addition to this, I'd suggest you leverage the `loading` prop on the `Button` component to further inform the User of the a loading coupled with the disabling.
> 
> It'll only complete the user experience.

Thank you for the thoughtful suggestion! 🙏🏼 Adding the loading prop to the Button component is indeed a fantastic idea to enhance the user experience.

**Review by bdougie — APPROVED** [review]
Thanks @deepakrudrapaul

**Comment by babblebey** [comment]
LG 🍕

**Review by babblebey — COMMENTED** [review]
LG 🍕

**Review by bdougie — APPROVED** [review]

**Review by brandonroberts — APPROVED** [review]

**Pull request merged**